data/transactions/logic/evalAppTxn_test.go

```diff
@@ -128,6 +128,71 @@ func TestFieldTypes(t *testing.T) {
 	TestApp(t, NoTrack("itxn_begin; byte \"pay\"; itxn_field Nonparticipation;"), ep, "not a uint64")
 }
 
+func TestFieldLimits(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	ep, _, _ := MakeSampleEnv()
+
+	intProgram := "itxn_begin; int %d; itxn_field %s; int 1"
+	goodInt := func(field string, value interface{}) {
+		TestApp(t, fmt.Sprintf(intProgram, value, field), ep)
+	}
+	badInt := func(field string, value interface{}) {
+		// error messages are different for different fields, just use a space
+		// to indicate there should be an error, it will surely match any error.
+		TestApp(t, NoTrack(fmt.Sprintf(intProgram, value, field)), ep, " ")
+	}
+	testInt := func(field string, max int) {
+		goodInt(field, 1)
+		goodInt(field, max)
+		badInt(field, max+1)
+	}
+	testBool := func(field string) {
+		goodInt(field, 0)
+		goodInt(field, 1)
+		badInt(field, 2)
+	}
+	bytesProgram := "itxn_begin; byte %#v; itxn_field %s; int 1"
+	goodBytes := func(field string, value string) {
+		TestApp(t, fmt.Sprintf(bytesProgram, value, field), ep)
+	}
+	badBytes := func(field string, value string) {
+		// error messages are different for different fields, just use a space
+		// to indicate there should be an error, it will surely match any error.
+		TestApp(t, NoTrack(fmt.Sprintf(bytesProgram, value, field)), ep, " ")
+	}
+	testBytes := func(field string, maxLen int) {
+		goodBytes(field, "")
+		goodBytes(field, strings.Repeat("a", maxLen))
+		badBytes(field, strings.Repeat("a", maxLen+1))
+	}
+
+	// header
+	badInt("TypeEnum", 0)
```

Comment:
0 for `UnknownTx`?

Reply:
Right, that's not allowed when using `itxn_field`